pyproject.toml

```
@@ -12,32 +12,50 @@ show_missing = true
 minversion = "6.0"
 log_cli_level = "INFO"
 
-# Formatting tools configuration
-[tool.black]
+# Linting and formatting tools configuration
+[tool.ruff]
+# Default settings: https://docs.astral.sh/ruff/configuration/
+# Settings: https://docs.astral.sh/ruff/settings/
 line-length = 99
-target-version = ["py38"]
+include = ["pyproject.toml", "src/**/*.py", "tests/**/*.py", "lib/charms/kratos/**/*.py"]
+extend-exclude = ["__pycache__", "*.egg_info"]
+target-version = "py38"
 
-[tool.isort]
-line_length = 99
-profile = "black"
+[too.ruff.lint]
+# Rules: https://docs.astral.sh/ruff/rules/
+select = ["E", "W", "F", "C", "N", "D", "I001"]
+ignore = [
+    "D100",
+    "D101",
+    "D102",
+    "D103",
+    "D105",
+    "D107",
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D400",
+    "D404",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+    "E501",
+    "N818"
+]
+per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
 
-# Linting tools configuration
-[tool.flake8]
-max-line-length = 99
-max-doc-length = 99
+[too.ruff.lint.flake8-copyright]
```

issue: watch out for the `too` typo for `tool`
```suggestion
[tool.ruff.lint.flake8-copyright]
```